src/CMakeLists.txt

```diff
@@ -7,9 +7,12 @@ add_library(mercury SHARED
 target_compile_options(mercury
 	PRIVATE -fprofile-arcs -ftest-coverage
 )
-target_link_options(mercury
-	PRIVATE -coverage
-)
+
+# option below is available starting with cmake 3.13
+# cmake on the pi is v3.7.2
+#target_link_options(mercury
+#	PRIVATE -coverage
+#)
```

Comment:
If we are disabling `-coverage` as a link option, I'm surprised that we don't also need to disable the `target_compile_options` above for `-fprofile-arcs -ftest-coverage` as they will require linkage options.

Reply:
What I noticed is that `lcov` was trying to obtain information as we discussed on our pair-programming session, but it failed at some later stage for reason still can't grasp for lack of knowledge on this. Suggest we adress all of this testing on a later PR